src/Tests/Tests/Test_FastConv.cpp

```diff
@@ -23,19 +25,158 @@ namespace fastconv_test {
     protected:
         void SetUp() override
         {
+            m_pCFastConv = new CFastConv();
+
+            m_IrLength = 51;
+            m_ImpulseLength = 10;
+            m_Ir = new float[m_IrLength];
+            m_Impulse = new float[m_ImpulseLength];
+
+            for (int i = 0; i < m_IrLength; i++)
+            {
+                if (i < m_ImpulseLength) { m_Impulse[i] = 0; };
+                m_Ir[i] = (rand() / static_cast<float>(RAND_MAX));
+                // m_Ir[i] = 10;
+            }
+            m_Impulse[3] = 1;
+            //m_Impulse[1] = 1;
```

Comment:
Also I like the definition of the impulses in this setup function! The only other thing I would add is the differently between "Ir" and "Impulse", maybe you should make it more clear since they mean the same thing to me I think your "Ir" is the random signal, and then your "Impulse" is the actual impulse with a singular 1 next to 0s? Maybe naming "Ir" to "signal" or "randIR" would be more clear for a user reading the tests/use case of the convolution